crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1623,6 +1624,119 @@ fn handle_named_expr_comment<'a>(
     }
 }
 
+/// Handles comments around the `:` token in a lambda expression.
+///
+/// For parameterized lambdas, both the comments between the `lambda` and the parameters, and the
+/// comments between the parameters and the body, are considered dangling, as is the case for all
+/// of the following:
+///
+/// ```python
+/// (
+///     lambda  # 1
+///     # 2
```

Comment:
What's the reason for making all comments dangling vs forcing line breaks when formatting the lambda if the argument e.g. has leading comments?

Reply:
Are you suggesting that `# 1` here should be a leading comment? Would you expect it to be moved to an own-line comment, or rendered as-is in this snippet?

Reply:
we could do it similar to lists, where 1 is dangling and 2 is leading:
```python
x = [  # 1
    # 2
    "a",
    "b",
    "c",
]
```

```
7..10, None, Some((ExprConstant, 23..26)), (ExprList, 4..47), "# 1"
15..18, None, Some((ExprConstant, 23..26)), (ExprList, 4..47), "# 2"
{
    Node {
        kind: ExprList,
        range: 4..47,
        source: `[  # 1⏎`,
    }: {
        "leading": [],
        "dangling": [
            SourceComment {
                text: "# 1",
                position: EndOfLine,
                formatted: true,
            },
        ],
        "trailing": [],
    },
    Node {
        kind: ExprConstant,
        range: 23..26,
        source: `"a"`,
    }: {
        "leading": [
            SourceComment {
                text: "# 2",
                position: OwnLine,
                formatted: true,
            },
        ],
        "dangling": [],
        "trailing": [],
    },
}
x = [  # 1
    # 2
    "a",
    "b",
    "c",
]
```